plugins/tracker-resources/src/components/SetDueDateActionPopup.svelte

```diff
@@ -30,7 +30,15 @@
     const newDueDate = detail && detail?.getTime()
 
     if (shouldSaveOnChange && newDueDate !== undefined && newDueDate !== value.dueDate) {
-      await client.update(value, { dueDate: newDueDate })
+      await client.updateCollection(
```

Comment:
It was not necessary, please check and update in next PR. client.update internally check for derived class for attached documents and automatically perform required changes.